datafusion/physical-optimizer/src/pruning.rs

```diff
@@ -1566,6 +1599,50 @@ fn build_predicate_expression(
         return expr;
     }
 
+    // Special handlng for floats. Because current Parquet statistics do not allow NaN, and
```

Comment:
This block is why we need to get the column ordering info passed down. Here we know which column is being pruned and with which operation. For floats we can disallow pruning because the Parquet stats are incomplete. We can also skip pruning if the stats are not valid because an ordering is not defined for the type.